include/zephyr/drivers/interrupt_controller/intc_vim.h

```diff
@@ -0,0 +1,383 @@
+#define VIM_GRP_STS_MSK_MAX                  (BIT_MASK(32))
+
+#define VIM_GRP_STS_RESETVAL                 (0x00000000U)
+
+/* INTR_EN_SET */
+
+#define VIM_GRP_INTR_EN_SET_MSK_MASK         (BIT_MASK(32))
+#define VIM_GRP_INTR_EN_SET_MSK_SHIFT        (0x00000000U)
+#define VIM_GRP_INTR_EN_SET_MSK_RESETVAL     (0x00000000U)
+#define VIM_GRP_INTR_EN_SET_MSK_MAX          (BIT_MASK(32))
+
+#define VIM_GRP_INTR_EN_SET_RESETVAL         (0x00000000U)
+
+/* INTR_EN_CLR */
+
+#define VIM_GRP_INTR_EN_CLR_MSK_MASK         (BIT_MASK(32))
+#define VIM_GRP_INTR_EN_CLR_MSK_SHIFT        (0x00000000U)
+#define VIM_GRP_INTR_EN_CLR_MSK_RESETVAL     (0x00000000U)
+#define VIM_GRP_INTR_EN_CLR_MSK_MAX          (BIT_MASK(32))
+
+#define VIM_GRP_INTR_EN_CLR_RESETVAL         (0x00000000U)
+
+/* IRQSTS */
+
+#define VIM_GRP_IRQSTS_MSK_MASK              (BIT_MASK(32))
+#define VIM_GRP_IRQSTS_MSK_SHIFT             (0x00000000U)
+#define VIM_GRP_IRQSTS_MSK_RESETVAL          (0x00000000U)
+#define VIM_GRP_IRQSTS_MSK_MAX               (BIT_MASK(32))
+
+#define VIM_GRP_IRQSTS_RESETVAL              (0x00000000U)
+
+/* FIQSTS */
+
+#define VIM_GRP_FIQSTS_MSK_MASK              (BIT_MASK(32))
+#define VIM_GRP_FIQSTS_MSK_SHIFT             (0x00000000U)
+#define VIM_GRP_FIQSTS_MSK_RESETVAL          (0x00000000U)
+#define VIM_GRP_FIQSTS_MSK_MAX               (BIT_MASK(32))
+
+#define VIM_GRP_FIQSTS_RESETVAL              (0x00000000U)
+
+/* INTMAP */
+
+#define VIM_GRP_INTMAP_MSK_MASK              (BIT_MASK(32))
+#define VIM_GRP_INTMAP_MSK_SHIFT             (0x00000000U)
+#define VIM_GRP_INTMAP_MSK_RESETVAL          (0x00000000U)
+#define VIM_GRP_INTMAP_MSK_MAX               (BIT_MASK(32))
+
+#define VIM_GRP_INTMAP_RESETVAL              (0x00000000U)
+
+/* INTTYPE */
+
+#define VIM_GRP_INTTYPE_MSK_MASK             (BIT_MASK(32))
+#define VIM_GRP_INTTYPE_MSK_SHIFT            (0x00000000U)
+#define VIM_GRP_INTTYPE_MSK_RESETVAL         (0x00000000U)
+#define VIM_GRP_INTTYPE_MSK_MAX              (BIT_MASK(32))
+
+#define VIM_GRP_INTTYPE_RESETVAL             (0x00000000U)
+
+/* INT */
+
+#define VIM_PRI_INT_VAL_MASK                 (BIT_MASK(4))
+#define VIM_PRI_INT_VAL_SHIFT                (0x00000000U)
+#define VIM_PRI_INT_VAL_RESETVAL             (BIT_MASK(4))
+#define VIM_PRI_INT_VAL_MAX                  (BIT_MASK(4))
+
+#define VIM_PRI_INT_RESETVAL                 (BIT_MASK(4))
+
+/* INT */
+
+#define VIM_VEC_INT_VAL_MASK                 (0xFFFFFFFCU)
+#define VIM_VEC_INT_VAL_SHIFT                (0x00000002U)
+#define VIM_VEC_INT_VAL_RESETVAL             (0x00000000U)
+#define VIM_VEC_INT_VAL_MAX                  (BIT_MASK(30))
+
+#define VIM_VEC_INT_RESETVAL                 (0x00000000U)
+
+/* PID */
+
+#define VIM_PID_MINOR_MASK                   (BIT_MASK(6))
+#define VIM_PID_MINOR_SHIFT                  (0x00000000U)
+#define VIM_PID_MINOR_RESETVAL               (0x00000001U)
+#define VIM_PID_MINOR_MAX                    (BIT_MASK(6))
+
```

Comment:
Also is this file derived from any of the existing TI SDK sources? looking at these VID/PID/RTL register definitions it looks like that.

Reply:
I have added all the register description available for VIM module. Took reference from PDK. Should I remove unused register definitions?

Reply:
Thanks for clarifying, I believe the implementation of VIM you have is feature complete, if you think you will need them in future you can keep it, otherwise let's remove unused ones.